test/siphash_internal_test.c

```diff
@@ -237,10 +225,13 @@ static int test_siphash(int idx)
     size_t i;
 
     if (expectedlen != SIPHASH_MIN_DIGEST_SIZE &&
-        expectedlen != SIPHASH_MAX_DIGEST_SIZE)
+        expectedlen != SIPHASH_MAX_DIGEST_SIZE) {
+        TEST_info("size %" OSSLzu " vs %d and %d", expectedlen,
+                  SIPHASH_MIN_DIGEST_SIZE, SIPHASH_MAX_DIGEST_SIZE);
```

Comment:
Yeah, there are plenty of places where limited use of C++ features would be a boon.
The risk is overusing/overdoing the C++ features and making a huge unmaintainable mess.

And writing templates is convoluted.  It seems to turn into a game of who can create the best error message longer than a page :(